Zombie/src/main/java/dk/gruppeseks/bodtrd/zombie/ZombieProcessor.java

```diff
@@ -72,63 +77,65 @@ public void process(World world)
             {
                 if (velocity.getMagnitude() <= ATTACK_RANGE + (zombieBod.getWidth() / 2) + (playerBod.getWidth() / 2))
                 {
-                    attackPlayer(world, ent, player);
+                    attackPlayer(ent, player);
                 }
                 velocity.setMagnitude(MOVEMENT_SPEED);
             }
             else if (velocity.getMagnitude() < AGGRO_RANGE)
             {
-                Path path = ent.get(Path.class);
-                if (path == null || (currentTime - path.getCreationTime()) > TIME_BETWEEN_PATH_UPDATE)
+                AIData aiDat = ent.get(AIData.class);
+
+                if (aiDat.getPath() == null || (currentTime - aiDat.getCreationTime()) > TIME_BETWEEN_PATH_UPDATE)
                 {
                     try
                     {
-                        Path newPath = _ai.getPath(zombieCenter, playerCenter, world);
-                        ent.add(newPath);
-                        velocity = new Vector2(zombieCenter, newPath.peekPosition());
+                        aiDat.setPath(_ai.getPath(zombieCenter, playerCenter, world));
+                        velocity = new Vector2(zombieCenter, aiDat.peekPosition());
                     }
                     catch (NoPathException ex)
                     {
                         velocity.setMagnitude(MOVEMENT_SPEED);
                     }
                 }
-                if (path != null)
+                if (aiDat.getPath() != null)
                 {
-                    Vector2 currentVector = new Vector2(zombieCenter, path.peekPosition());
+                    Vector2 currentVector = new Vector2(zombieCenter, aiDat.peekPosition());
                     if (currentVector.getMagnitude() < 10) // Currently the fact that their position isn't at center is a problem, because sometimes it goes under 851-849.99 for example and it thinks its on the other grid. etc etc
                     {
-                        if (path.size() > 1)
+                        if (aiDat.size() > 1)
                         {
-                            path.popPosition();
+                            aiDat.popPosition();
```

Comment:
imo this doesnt make any sense, i was thinking that Path be a class contained by AIData